pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -844,12 +843,12 @@ public LLRealtimeSegmentDataManager(RealtimeSegmentZKMetadata segmentZKMetadata,
     _fieldExtractor = FieldExtractorFactory.getPlainFieldExtractor(schema);
     makeConsumerWrapper();
 
-    // Partitioners
-    _column2PartitionersMap = indexingConfig.getPartitioners();
-    if (_column2PartitionersMap != null && !_column2PartitionersMap.isEmpty()) {
+    SegmentPartitionConfig segmentPartitionConfig = indexingConfig.getSegmentPartitionConfig();
+    if (segmentPartitionConfig != null) {
       try {
         int nPartitions = _consumerWrapper.getPartitionCount(_kafkaTopic, /*maxWaitTimeMs=*/5000L);
-        _realtimeSegment.setSegmentPartitionConfig(createPartitionConfig(nPartitions));
+        segmentPartitionConfig.setNumPartitions(nPartitions);
+        _realtimeSegment.setSegmentPartitionConfig(segmentPartitionConfig);
       } catch (Exception e) {
         segmentLogger.warn("Couldn't get number of partitions in 5s, not using partition config");
```

Comment:
can u please add e.getMessage() to this warning? thanks. segmentLogger automatically adds segmentName (that has table name and partition ID)